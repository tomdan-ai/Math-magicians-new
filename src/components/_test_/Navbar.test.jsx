import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navebar from '../Navbar';

describe('Navbar', () => {
  test('Navbar is being rendered correctly', () => {
    const container = document.createElement('div');
    render(
      <BrowserRouter basename="/">
        <Navebar />
      </BrowserRouter>,
      container,
    );
    const navElement = screen.getByRole('navigation');
    expect(navElement).toBeInTheDocument();
  });
});

describe('Navbar Snapshot', () => {
  test('Navbar matches DOM snapshot', () => {
    const tree = renderer.create(
      <BrowserRouter basename="/">
        <Navebar />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
