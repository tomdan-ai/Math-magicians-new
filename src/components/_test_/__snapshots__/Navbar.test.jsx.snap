// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Snapshot Navbar matches DOM snapshot 1`] = `
<div>
  <h1
    className="header"
  >
    Math magicians
  </h1>
  <nav
    className="navbar"
  >
    <a
      className="home"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="quotes"
      href="/Quotes"
      onClick={[Function]}
    >
      Quotes
    </a>
    <a
      className="calculator"
      href="/Calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
  </nav>
</div>
`;
